src/libs/ReportActionsUtils.ts

```diff
@@ -39,194 +45,136 @@ Onyx.connect({
     return childType === CONST.REPORT.TYPE.CHAT && (childVisibleActionCount > 0 || String(childReportID) === reportID);
 }
 
 /**
  * Returns the parentReportAction if the given report is a thread/task.
  *
- * @param {Object} report
- * @param {Object} [allReportActionsParam]
- * @returns {Object}
  * @deprecated Use Onyx.connect() or withOnyx() instead
  */
-function getParentReportAction(report, allReportActionsParam = undefined) {
-    if (!report || !report.parentReportID || !report.parentReportActionID) {
+function getParentReportAction(report: OnyxEntry<OnyxTypes.Report>, allReportActionsParam?: OnyxCollection<OnyxTypes.ReportActions>): OnyxTypes.ReportAction | Record<string, never> {
+    if (!report?.parentReportID || !report.parentReportActionID) {
         return {};
     }
-    return lodashGet(allReportActionsParam || allReportActions, [report.parentReportID, report.parentReportActionID], {});
+    return (allReportActionsParam ?? allReportActions)?.[report.parentReportID]?.[report.parentReportActionID] ?? {};
 }
 
 /**
  * Determines if the given report action is sent money report action by checking for 'pay' type and presence of IOUDetails object.
- *
- * @param {Object} reportAction
- * @returns {Boolean}
  */
-function isSentMoneyReportAction(reportAction) {
-    return (
-        reportAction &&
-        reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU &&
-        lodashGet(reportAction, 'originalMessage.type') === CONST.IOU.REPORT_ACTION_TYPE.PAY &&
-        _.has(reportAction.originalMessage, 'IOUDetails')
-    );
+function isSentMoneyReportAction(reportAction: OnyxEntry<OnyxTypes.ReportAction>): boolean {
+    return reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && reportAction.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.PAY && !!reportAction.originalMessage.IOUDetails;
 }
 
 /**
  * Returns whether the thread is a transaction thread, which is any thread with IOU parent
  * report action from requesting money (type - create) or from sending money (type - pay with IOUDetails field)
- *
- * @param {Object} parentReportAction
- * @returns {Boolean}
  */
-function isTransactionThread(parentReportAction) {
-    const originalMessage = lodashGet(parentReportAction, 'originalMessage', {});
+function isTransactionThread(parentReportAction: OnyxEntry<OnyxTypes.ReportAction>): boolean {
     return (
-        parentReportAction &&
-        parentReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU &&
-        (originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.CREATE || (originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.PAY && _.has(originalMessage, 'IOUDetails')))
+        parentReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU &&
+        (parentReportAction.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.CREATE ||
+            (parentReportAction.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.PAY && !!parentReportAction.originalMessage.IOUDetails))
```

Comment:
Logic was changed here on purpose? `originalMessage.type` -> `parentReportAction.originalMessage.type`

Reply:
Actually there is no logic change, here Fabio just removed `const originalMessage = lodashGet(parentReportAction, 'originalMessage', {});`

and just use `parentReportAction.originalMessage` etc